docs/architecture/provisioning-api.md

```diff
@@ -0,0 +1,120 @@
+# Introduction
+
+The goal of this document is to describe requirements for Provisioner component and propose an API. 
+
+## Requirements
+
+## Non-Functional
+
+The basic requirements are defined as follows:
+
+- Provisioner API is based on GraphQL. 
+- Provisioner API is asynchronous.
+
+## Functional
+
+### Runtime management
+
+The following operations must be supported:
+
+- Provisioning
+- Upgrade
+  - Kubernetes version.
+  - Kyma version.
+- Deprovisioning
+
+### Runtime Agent connection management
+
+Resetting connection between Runtime and Management Plane Services must be supported. The goal of this functionality is to force Compass Runtime Agent installed on the Runtime to establish a new connection.
```

Comment:
```suggestion
The API must support resetting the connection between Runtime and Management Plane Services. The goal of this functionality is to force Compass Runtime Agent installed on the Runtime to establish new connection.
```
I'd stick with one:
 either `resetting connection` and `establish new connection`
or 
`resetting the connection` and `to establish a new connection`.